petastorm/spark/spark_dataset_converter.py

```diff
@@ -303,10 +303,34 @@ def _convert_precision(df, precision):
     return df
 
 
+def _convert_vector(df, precision):
+    from pyspark.ml.linalg import Vectors, SparseVector
+    from pyspark.mllib.linalg import Vector as OldVector
+
+    types_set = {struct_field.dataType for struct_field in df.schema}
+    found_vectors = not types_set.isdisjoint({Vectors, SparseVector, OldVector})
+    if not found_vectors:
+        return df
+
+    import pyspark
+    if pyspark.__version__ < "3.0.0":
+        raise ValueError("Vector columns are not supported for pyspark<3.0.0.")
+    # pylint: disable=import-error
+    from pyspark.ml.functions import vector_to_array
+    # pylint: enable=import-error
+
+    for struct_field in df.schema:
+        col_name = struct_field.name
+        if struct_field.dataType in {Vectors, SparseVector, OldVector}:
```

Comment:
Vectors is not the vector base class.

Use `from pyspark.ml.linalg import Vector`
Remove importing SparseVector class (why do you need it here ?)

Reply:
Right, we don't support SparseVector now.